### PR TITLE
PFS load testing for new storage layer

### PR DESCRIPTION
**Pull request by brycemcanally**
This PR will contain load tests and load testing utilities for PFS (with a focus on the new storage layer).

The load testing utilities are designed to make the process of creating a load test composable. As of now, there are essentially four levels of composition: branch, commit, put tar, and file. Each level has a generator that takes generators for the level below. The idea here is that we can have different generators at each level that can be mixed and matched in different workloads. This should remove a lot of the boilerplate when creating load tests, which should make it easier to quickly make and maintain the load tests over time. An important part of the load testing effort for the new storage layer is to cover a large variety of workloads, so composable load tests seem like a good way to accomplish this.

### CODE REVIEW

File: src/server/pkg/storage/fileset/index/writer.go [hunk 1]
```diff
@@ -151,12 +151,20 @@ func (w *Writer) Close() error {
 	if err != nil {
 		return err
 	}
-	if _, err := pbutil.NewWriter(objW).Write(w.root); err != nil {
+	defer func() {
+		if retErr == nil {
```
Comment (jdoliner) [inline src/server/pkg/storage/fileset/index/writer.go:155]: This seems like a bug in that if we encounter another error we won't call `objW.Close()` and will keep an object write open.

Comment (brycemcanally) [reply src/server/pkg/storage/fileset/index/writer.go:155]: That is what we want. We want the object storage provider to timeout the write and delete what was written. If we close it, the data written will be kept. There might be a better pattern for this though that is not as weird (and potentially problematic for certain object storage providers).

Comment (jdoliner) [reply src/server/pkg/storage/fileset/index/writer.go:155]: I see, this makes sense, I think there is a better pattern here which is to create a `ctx` for the `Writer` and then explicitly call `cancel` if there's an error, that'll free up the resources immediately, and make it a little more clear that not calling `Close` is intentional in that case.

Comment (brycemcanally) [reply src/server/pkg/storage/fileset/index/writer.go]: After thinking about this more, and looking at the object storage clients, I decided to go ahead and always close. The object storage clients handle this case correctly. Also, when we change the upload interface to operate on a passed in reader, this potential problem will go away.

**Pull request merged**